core/payment/src/api/allocations.rs

```diff
@@ -50,6 +52,42 @@ async fn create_allocation(
         .unwrap_or(DEFAULT_PAYMENT_PLATFORM.to_string());
     let address = allocation.address.clone().unwrap_or(node_id.to_string());
 
+    // If the request contains information about the payment platform, initialize the account
+    // by setting the `send` field to `true`, as it is implied by the intent behing allocation of funds.
+    if let Some(platform) = &allocation.payment_platform {
+        // payment_platform is of the form driver-network-token
+        // eg. erc20-rinkeby-tglm
+        let [driver, network, token]: [&str; 3] =
+            match platform.split("-").collect::<Vec<_>>().try_into() {
+                Ok(arr) => arr,
+                Err(_e) => {
+                    return response::bad_request(
+                        &"paymentPlatform must be of the form driver-network-token",
+                    )
+                }
+            };
+
+        // The token needs to be uppercase excluding the initial 't', eg. tGLM
+        let token_no_prefix = match token.strip_prefix("t") {
+            Some(tok) => tok,
+            None => return response::bad_request(&"token must begin with 't'"),
```

Comment:
Only testnet tokens begin with `t`. Mainnet doesn't have this prefix.

Reply:
This is crucial, because it disables `mainnet` ;)

Reply:
I've dropped token parsing and made it default to `None`.